db/pushdb.go

```diff
@@ -145,11 +146,42 @@ func (f *pushDatabaseOpts) AddPushServiceProviderToService(service string,
 	}
 	f.dblock.Lock()
 	defer f.dblock.Unlock()
+
+	//#198:
+	//2017 Victor Lang
+	//Before add a new psp to service, try to verify there is no redundant psp in same type exists under the same service.
+	//Currently, redundant psp to service will result in problem in subscribe and push later.
+	expsps, err := f.db.GetPushServiceProvidersByService(service)
+	if err != nil {
+		return fmt.Errorf("Error querying psp with name: %v", err)
+	}
+	if len(expsps) > 0 {
+		for index := range expsps {
+			pushpsp, perr := f.db.GetPushServiceProvider(expsps[index])
+
+			if perr != nil {
+				return fmt.Errorf("Error to retrieve psp with name: %v", err)
+			}
+			//Whether the existing psp has the same push service type
+			if pushpsp.PushServiceName() == push_service_provider.PushServiceName() {
+				//The service has a psp in same push service type
```

Comment:
Convention is one space after // (before comment text)

Reply:
done